src/core/namespace.c

```diff
@@ -724,6 +793,20 @@ static int mount_empty_dir(const MountEntry *m) {
         return 1;
 }
 
+static int mount_tmpfs(const MountEntry *m) {
+        assert(m);
+
+        /* First, get rid of everything that is below if there is anything. Then, overmount with our new empty dir */
+
+        (void) mkdir_p_label(mount_entry_path(m), 0755);
+        (void) umount_recursive(mount_entry_path(m), 0);
+
+        if (mount("tmpfs", mount_entry_path(m), "tmpfs", MS_NODEV|MS_STRICTATIME, mount_entry_option(m)) < 0)
```

Comment:
Hmm.

Will the kernel help us by converting "noatime" into MS_NOATIME and override the constant we passed?

Similar question with "ro", "nosuid" etc.

If not, maybe libmount counld help us.

Reply:
It *might* not be good to rely on the raw kernel behaviour. Particularly if no-one else exercises it. If we are able to use libmount, and there is a weird problem, I think it will be more straightforward for the user to debug because they can compare more directly with the behaviour of "/bin/mount".

https://twitter.com/mjg59/status/920496374549856256

That's just a suggestion though, I haven't inspected the kernel codepath for when only tmpfs is used, to see how trustworthy or not I think it is :).